vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/CookieStoreImpl.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2011-2018 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.ext.web.client.impl;
+
+import java.util.Map.Entry;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.function.Consumer;
+
+import io.netty.handler.codec.http.cookie.Cookie;
+import io.netty.handler.codec.http.cookie.DefaultCookie;
+import io.vertx.ext.web.client.CookieStore;
+
+/**
+ * @author <a href="mailto:tommaso.nolli@gmail.com">Tommaso Nolli</a>
+ */
+public class CookieStoreImpl implements InternalCookieStore {
+
+  private ConcurrentHashMap<Key, Cookie> noDomainCookies;
+  private ConcurrentSkipListMap<Key, Cookie> domainCookies;
+  
+  public CookieStoreImpl() {
+    noDomainCookies = new ConcurrentHashMap<>();
+    domainCookies = new ConcurrentSkipListMap<>();
+  }
+  
+  @Override
+  public Iterable<Cookie> get(boolean ssl, String domain, String path) {
+    /* TODO: Shall we cleanup path (remove dots)? Or HttpRequestImpl.uri is always normalized?
+     *       Ideally this should call io.vertx.ext.web.impl.Utils.removeDots
```

Comment:
I think we should move it to core, as we already did for other utils method. I believe there is also such functionality in core but not exactly the same behavior.